app/src/test/java/ch/epfl/culturequest/map_collectiontest/BasicOTMProviderTest.java

```diff
@@ -0,0 +1,106 @@
+package ch.epfl.culturequest.map_collectiontest;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.containsInAnyOrder;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertThrows;
+
+import com.google.android.gms.maps.model.LatLng;
+
+
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.TimeUnit;
+
+import ch.epfl.culturequest.backend.map_collection.BasicOTMProvider;
+import ch.epfl.culturequest.backend.map_collection.OTMLocation;
+import ch.epfl.culturequest.backend.map_collection.OTMProvider;
+import okhttp3.mockwebserver.MockResponse;
+import okhttp3.mockwebserver.MockWebServer;
+
+public class BasicOTMProviderTest {
+    MockWebServer server = new MockWebServer();
+    OTMProvider provider = new BasicOTMProvider("http://localhost:8080/");
+
+    @Before
+    public void setUp() throws IOException {
+        server.start(8080);
+    }
+
+    // Test if the provider returns a failure if the server is not reachable
+    @Test
+    public void testProviderReturnsFailureIfServerIsNotReachable() throws IOException {
+        server.shutdown(); // make the server unavailable
+        assertThrows(CompletionException.class, () -> provider.getLocations(new LatLng(1., 0.), new LatLng(0.,1.)).orTimeout(5, TimeUnit.SECONDS).join());
+    }
```

Comment:
You shouldn't shut down the mock web server in your test to simulate the unavailability. I believe it actually breaks all the upcoming tests since the server isn't restarted afterwards.

Instead, I suggest you do:

```suggestion
    @Test
    public void testProviderReturnsFailureIfServerIsNotReachable() throws IOException {
        server.enqueue(new MockResponse().setSocketPolicy(SocketPolicy.DISCONNECT_AT_START));
        assertThrows(CompletionException.class, () -> provider.getLocations(new LatLng(1., 0.), new LatLng(0.,1.)).orTimeout(5, TimeUnit.SECONDS).join());
    }
```

Or, to simulate a timeout, you can also do:

`server.enqueue(new MockResponse().setResponseCode(200).setBodyDelay(10, TimeUnit.SECONDS));`

Reply:
This should not be a problem as I use a @Before method which restarts the server before each test. But I guess I can do that as well, it should not change anything